core/src/executor/sort.rs

```diff
@@ -41,48 +48,100 @@ impl<'a> Sort<'a> {
+                                    } => {
+                                        let value: Value =
+                                            evaluate_stateless(None, expr)?.try_into()?;
 
-                        async move {
-                            evaluate(self.storage, context, aggregated, expr)
-                                .await?
-                                .try_into()
-                                .map(|value| (value, *asc))
-                        }
-                    })
-                    .try_collect::<Vec<_>>()
-                    .await?;
+                                        match value {
+                                            Value::I64(_) => {
+                                                let index: usize = value.try_into()?;
+                                                let zero_based =
+                                                    index.checked_sub(1).ok_or_else(|| {
+                                                        crate::result::Error::from(
+                                                            SortError::ColumnIndexOutOfRange(index),
+                                                        )
+                                                    })?;
+
+                                                let value =
+                                                    row.get_value(zero_based).ok_or_else(|| {
+                                                        crate::result::Error::from(
+                                                            SortError::ColumnIndexOutOfRange(index),
+                                                        )
+                                                    })?;
 
-                Ok((values, aggregated, blend_context))
+                                                Ok::<_, Error>((value.clone(), *asc))
+                                            }
+                                            _ => Err(SortError::Unreachable.into()),
+                                        }
+                                    }
+                                    _ => {
+                                        let value: Value =
+                                            evaluate(self.storage, context, aggregated, expr)
+                                                .await?
+                                                .try_into()?;
+
+                                        Ok::<_, Error>((value, *asc))
+                                    }
+                                }
+                            }
+                        })
+                        .try_collect::<Vec<_>>()
+                        .await?;
+
+                    Ok((values, row))
+                }
             })
             .try_collect::<Vec<_>>()
             .await
             .map(Vector::from)?
-            .sort_by(|(values_a, _, _), (values_b, _, _)| {
+            .sort_by(|(values_a, ..), (values_b, ..)| {
```

Comment:
Wow, I didn't know that `..` also works in tuples. Looks better!

Reply:
lol but it's a bit weird that 
```rs
.map(|(.., row)| Ok(row));
```
works but
```rs
.map(|(row, ..)| Ok(row));
```
doesn't 🤣 
https://github.com/gluesql/gluesql/pull/805/files#diff-71b79d5c3dcf3ec4810d6e9149f90c16fff0dae2a728aa95f670d742baafba81R146

Reply:
There still exists the thing the Rust compiler can be smarter 😁